pil-analyzer/src/lib.rs

```diff
@@ -29,27 +29,32 @@ pub use pil_analyzer::{analyze_ast, analyze_file, analyze_string};
 
 pub trait AnalysisDriver: Clone + Copy {
     /// Turns a declaration into an absolute name.
-    fn resolve_decl(&self, name: &String) -> String {
+    fn resolve_decl(&self, name: &String) -> Result<String, String> {
         self.resolve_namespaced_decl(&[name])
-            .relative_to(&Default::default())
-            .to_string()
+            .map(|path| path.relative_to(&Default::default()).to_string())
     }
     /// Turns a nested declaration into an absolute name.
-    fn resolve_namespaced_decl(&self, path: &[&String]) -> AbsoluteSymbolPath;
-    fn resolve_value_ref(&self, path: &SymbolPath) -> String {
+    fn resolve_namespaced_decl(&self, path: &[&String]) -> Result<AbsoluteSymbolPath, String>;
```

Comment:
Actually I think this is not supposed to fail because it does not do a lookup, it just makes the ref absolute.